frontend/setupChelonia.js

```diff
@@ -273,6 +275,25 @@ const setupChelonia = async (): Promise<*> => {
     })
   })
 
+  sbp('okTurtles.events/on', CONTRACTS_MODIFIED, (_, { added }) => {
+    // Wait for the added contracts to be ready, then call the update function
+    if (!added.length) return
+    const rootState = sbp('chelonia/rootState')
+    added.forEach((cID) => {
+      switch (rootState.contracts[cID]?.type) {
+        case 'gi.contracts/identity':
+          if (cID === rootState.loggedIn?.identityContractID) {
+            sbp('chelonia/contract/setKvFilter', cID, [KV_KEYS.UNREAD_MESSAGES, KV_KEYS.PREFERENCES, KV_KEYS.NOTIFICATIONS])
+          }
+          break
```

Comment:
This used to say `return`, which I think is correct, as now we will first set the filter for `cID` to `[KV_KEYS.UNREAD_MESSAGES, KV_KEYS.PREFERENCES, KV_KEYS.NOTIFICATIONS]` and then overwrite that with `[]` down below.

Reply:
I think what you want is a `return` inside of the `if`, and a `break` here as you have it.

Reply:
Yes, that sounds about right. The way it was before it was setting no filter on other identity contracts